### PR TITLE
Displaying the IRC users in the Widelands-Lobby

### PR DESCRIPTION
**Pull request by Notabilis**
IRC nicknames are displayed in the Widelands lobby as users with a build version of "IRC".

Handles the metaserver-side of issue #3.

This branch probably shouldn't be merged/deployed immediately. With a low window size / resolution the IRC users fill the player list, forcing the user to use the scrollbar to see the "real" players in the lobby. On the side of the Widelands client a button to hide them might be an idea, and/or listing the IRC users at the bottom of the list by default.

### CODE REVIEW

**Comment by SirVer** [comment]
Listing the IRC users at the bottom (maybe with a different/darker color) sounds good. No harm in reviewing and merging this branch I think - as soon as we deploy we annoy all b19 players anyways, right?

File: wlms/ircbridge.go [hunk 1]
```diff
@@ -2,12 +2,12 @@ package main
...
 import (
 	"log"
-
 	"github.com/thoj/go-ircevent"
+	"strings"
 )
 
 type IRCBridger interface {
-	Connect(chan Message, chan Message) bool
+	Connect(chan Message, chan Message, chan string, chan string) bool
```
Comment (SirVer) [inline wlms/ircbridge.go]: add documentation on the parameters. They become unwieldy.

Also, can you instead pass in a struct that has named members?

File: wlms/ircbridge.go [hunk 2]
```diff
@@ -54,10 +54,41 @@ func (bridge *IRCBridge) Connect(messagesIn chan Message, messagesOut chan Messa
 			message: event.Message(),
 		}:
 		default:
-			log.Println("Message Queue full.")
+			log.Println("IRC Message Queue full.")
+		}
+	})
+	bridge.connection.AddCallback("JOIN", func(e *irc.Event) {
+		if e.Nick != bridge.nick {
```
Comment (SirVer) [inline wlms/ircbridge.go]: favor early return to avoid right drift: `if e.Nick == bridge.nick {\nreturn;\n}`. Same below.

**Comment by SirVer** [comment]
One general question: Can we hide the IRC users from b19 and below? to avoid spamming them in the UI with names that they cannot use for anything.

**Comment by Notabilis** [comment]
Thanks for the review.
Admittedly I forgot about b19. Hiding the IRC users should be easily possible, good idea.
Sorting the users on the client shouldn't be any problem either. No idea whether a different color is possible, I have to look into that.

**Comment by Notabilis** [comment]
Review remarks should be done and build19 users are no longer spammed with IRC entries.
As far as I am concerned, this branch can be merged and deployed now.

I proposed a branch for sorting the users in the lobby:
https://code.launchpad.net/~widelands-dev/widelands/irc_users/+merge/335898

File: wlms/ircbridge.go [hunk 3]
```diff
@@ -6,8 +6,29 @@ import (
 	"strings"
 )
 
+// Structure with channels for communication between IRCBridger and the metaserver
```
Comment (SirVer) [inline wlms/ircbridge.go]: typo? IRCBridger. Should that not be IRCBridge everywhere?

**Review by SirVer — APPROVED** [review]

**Comment by SirVer** [comment]
One question, otherwise lgtm. Feel free to merge and deploy when you feel like it.

File: wlms/ircbridge.go (hunk 3 above)
Comment (Notabilis) [reply wlms/ircbridge.go]: At least no typo of mine. For some reason the interface class of the IRC bridge is named that way. IRCBridger is the interface, IRCBridge the struct/class which implements it...

**Comment by Notabilis** [comment]
Okay, thanks. Will do so tomorrow.

**Pull request merged**